test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -869,15 +869,45 @@ public ClusterHealthStatus ensureGreen(String... indices) {
      * @param timeout time out value to set on {@link org.elasticsearch.action.admin.cluster.health.ClusterHealthRequest}
      */
     public ClusterHealthStatus ensureGreen(TimeValue timeout, String... indices) {
-        ClusterHealthResponse actionGet = client().admin().cluster()
-            .health(Requests.clusterHealthRequest(indices).timeout(timeout).waitForGreenStatus().waitForEvents(Priority.LANGUID).waitForNoRelocatingShards(true)).actionGet();
+        return ensureGreenOrYellow(true, timeout, indices);
+    }
+
+    /**
+     * Ensures the cluster has a yellow state via the cluster health API.
+     */
+    public ClusterHealthStatus ensureYellow(String... indices) {
+        return ensureGreenOrYellow(false, TimeValue.timeValueSeconds(30), indices);
+    }
+
+    private ClusterHealthStatus ensureGreenOrYellow(boolean green, TimeValue timeout, String... indices) {
+        String color = green ? "green" : "yellow";
+        String method = green ? "ensureGreen" : "ensureYellow";
+
+        ClusterHealthRequest healthRequest = Requests.clusterHealthRequest(indices)
+            .timeout(timeout)
+            .waitForEvents(Priority.LANGUID)
+            .waitForNoRelocatingShards(true);
+        if (green) {
+            healthRequest.waitForGreenStatus();
+        } else {
+            healthRequest.waitForYellowStatus();
+        }
+        if (cluster() != null) {
```

Comment:
why is this checked? `client()` needs a cluster to be there, so it should always be non null?